cmd/queue/main.go

```diff
@@ -201,8 +205,57 @@ func createAdminHandlers() *http.ServeMux {
 	return mux
 }
 
+func probeQueueHealthPath(port string) error {
+	url := fmt.Sprintf("http://127.0.0.1:%s%s", port, queue.RequestQueueHealthPath)
+
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			// Do not use the cached connection
+			DisableKeepAlives: true,
+		},
+		Timeout: 500 * time.Millisecond,
+	}
+
+	var lastErr error
+
+	err := wait.PollImmediate(100*time.Millisecond, time.Second, func() (bool, error) {
+		var res *http.Response
+		res, lastErr = httpClient.Get(url)
+
+		if res == nil {
+			return false, nil
+		}
+
+		defer res.Body.Close()
+		return res.StatusCode == http.StatusOK, nil
+	})
+
+	if lastErr != nil {
+		return fmt.Errorf("failed to probe: %s", lastErr)
```

Comment:
This should use `github.com/pkg/errors` and do `errors.Wrap(...)`.

Reply:
👍